runtime/doc/lsp.txt

```diff
@@ -0,0 +1,359 @@
+*lsp.txt* The Language Server Protocol
+
+        NVIM REFERENCE MANUAL
+
+
+Neovim Language Server Protocol (LSP) API
+
+Neovim exposes a powerful API that conforms to Microsoft's published Language
+Server Protocol specification. The documentation can be found here:
+
+	https://microsoft.github.io/language-server-protocol/
+
+
+================================================================================
+                                                            *lsp-server-config*
+
+Neovim only exposes a powerful API for the language server protocol. To get
+the real benefits of this API, a language server must be installed. Many
+examples can be found here:
+
+	https://microsoft.github.io/language-server-protocol/implementors/servers/
+
+After installing a language server to your machine, you must let Neovim know
+how to start and interact with that language server. To do so, use the
+following APIs:
+
+                                                      *lsp#add_server_config()*
+lsp#add_server_config({config}):
+	Used to add a language server.
+
+	This will add an autocmd that will start the server and handle any other
+	startup activities whenever the |FileType| event is triggered for {ftype}.
+
+	Parameters: ~
+		{config} (dictionary): 
+			`filetype` (string|array):
+				A string or list of strings of filetypes to
+				associate with this server.
+
+			`server_name` (string) (optional):
+				You can pass any name as server_name.
+				If you want to use multiple language servers
+				for one filetype, be sure to add server_name.
+
+			`cmd` (dictionary): Executing language server command.
+				`execute_path` (string):
+					Executing language server command path.
+				`args` (array):
+					Command arguments.
+
+			`server_config` (dictionary): Optional configuration for a server.
+
+	Example: ~
+>
+		" C, C++ example
+		call lsp#add_server_config({
+		      \ 'filetype': ['c', 'cpp'],
+		      \ 'server_name': 'cland',
+		      \ 'cmd': { 'execute_path': 'cland', 'args': ['-background-index'] }})
+<
+>
+		" Go example
+		call lsp#add_server_config({
+		      \ 'filetype': 'go',
+		      \ 'server_name': 'gopls',
+		      \ 'cmd': { 'execute_path': 'gopls', 'args': [] }})
+<
+>
+		" Python example
+		call lsp#add_server_config({
+		      \ 'filetype': 'python',
+		      \ 'server_name': 'pyls',
+		      \ 'cmd': { 'execute_path': 'pyls', 'args': [] }})
+<
+>
+		" Rust example
+		call lsp#add_server_config({
+		      \ 'filetype': 'rust',
+		      \ 'server_name': 'rls',
+		      \ 'cmd': { 'execute_path': 'rustup', 'args': ['run', 'stable', 'rls'] },
```

Comment:
Isn't just `['rls']` better (and use whatever rustup channel that is active for the current environment)?